--- a/client.c
+++ b/client.c
@@ -295,19 +295,6 @@
 
 	if (r->expect_cont)
 		ustream_printf(cl->us, "HTTP/1.1 100 Continue\r\n\r\n");
-
-	switch(r->ua) {
-	case UH_UA_MSIE_OLD:
-		if (r->method != UH_HTTP_MSG_POST)
-			break;
-
-		/* fall through */
-	case UH_UA_SAFARI:
-		r->connection_close = true;
-		break;
-	default:
-		break;
-	}
 
 	uh_handle_request(cl);
 }
@@ -355,34 +342,6 @@
 	} else if (!strcmp(data, "connection")) {
 		if (!strcasecmp(val, "close"))
 			r->connection_close = true;
-	} else if (!strcmp(data, "user-agent")) {
-		char *str;
-
-		if (strstr(val, "Opera"))
-			r->ua = UH_UA_OPERA;
-		else if ((str = strstr(val, "MSIE ")) != NULL) {
-			r->ua = UH_UA_MSIE_NEW;
-			if (str[5] && str[6] == '.') {
-				switch (str[5]) {
-				case '6':
-					if (strstr(str, "SV1"))
-						break;
-					/* fall through */
-				case '5':
-				case '4':
-					r->ua = UH_UA_MSIE_OLD;
-					break;
-				}
-			}
-		}
-		else if (strstr(val, "Chrome/"))
-			r->ua = UH_UA_CHROME;
-		else if (strstr(val, "Safari/") && strstr(val, "Mac OS X"))
-			r->ua = UH_UA_SAFARI;
-		else if (strstr(val, "Gecko/"))
-			r->ua = UH_UA_GECKO;
-		else if (strstr(val, "Konqueror"))
-			r->ua = UH_UA_KONQUEROR;
 	}
 
 
--- a/uhttpd.h
+++ b/uhttpd.h
@@ -109,22 +109,9 @@
 	UH_HTTP_VER_1_1,
 };
 
-enum http_user_agent {
-	UH_UA_UNKNOWN,
-	UH_UA_GECKO,
-	UH_UA_CHROME,
-	UH_UA_SAFARI,
-	UH_UA_MSIE,
-	UH_UA_KONQUEROR,
-	UH_UA_OPERA,
-	UH_UA_MSIE_OLD,
-	UH_UA_MSIE_NEW,
-};
-
 struct http_request {
 	enum http_method method;
 	enum http_version version;
-	enum http_user_agent ua;
 	int redirect_status;
 	int content_length;
 	bool expect_cont;
